[14.0][IMP] cetmix_tower_yaml: New models and fields, export wizard update

ivs-cetmix opened the pull request:
**Variables**

- Variable Type
- Options

**Keys**

- Don't export secret value, remove obsolete methods.

**Server Template**

- Export 'On Delete Plan'.

**Export Wizard**

- Option to remove empty x2m field values. This is useful for keeping the YAML file smaller and cleaner.
- Improve form layout for better UX.

Task: 4219

# PR requirements

## Commit message

### Title (subject line)

Short and descriptive. Please try to keep it under 50 characters
Written in the imperative mood (e.g., "Fix bug", "Add feature" **not** "Fixed" or "Added").

Use tags listen in the [Odoo Guidelines](https://www.odoo.com/documentation/16.0/contributing/development/git_guidelines.html#tag-and-module-name).

### ​Description (mandatory)

- Must be separated from the title with a blank line.
- Wrap at 72 characters per line for readability.
- Must explain **why** and **what** of the change (if not self-explanatory).
- Can mention **how** **only if necessary** for clarification.
- Must have a line with related task number. Eg "Task: 2302"

Commit message example

```txt
[ADD] cetmix_tower_server: Doge memes

Implement Doge memes in order to make the server more fun.

Add new models to allow more flexible meme configuration:

- Doge
- Meme
- DogeMeme

```

## PR description

Provide Odoo version number tag together with the primary tag. Example:

```txt
[18.0][ADD] cetmix_tower_server: Doge memes
```

PR description should contain context and description of the changes:

```txt
Implement Doge memes in order to make the server more fun.
Add new models to allow more flexible meme configuration:

- Doge
- Meme
- DogeMeme
```

NB: Please ensure that you have signed the CLA. There will be a message in the PR review if you have not.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
	- Added option to remove empty values during YAML export.
	- Added file header to YAML export.
	- Enhanced variable and variable option YAML representation.

- **Refactor**
	- Removed secret value and SSH password masking logic.
	- Updated record processing for YAML operations.
	- Introduced a new class for variable options.

- **Tests**
	- Added test for YAML export without empty values.
	- Removed tests for secret value and SSH password masking.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/cetmix/cetmix-tower?pullRequest=193) <br/>All committers have signed the CLA.

Aldeigja reviewed (approved):
Functional LGTM

ivs-cetmix committed:
[IMP] cetmix_tower_yaml: New models and fields

**Variables**

- Variable Type
- Options

**Keys**

- Don't export secret value, remove obsolete methods.

**Server Template**

- Export 'On Delete Plan'.

**Export Wizard**

- Option to remove empty x2m field values. This is useful for keeping
  the YAML file smaller and cleaner.
- Improve form formatting for better UX.

Task: 4219

ivs-cetmix commented:
/ocabot merge nobump

CetmixGitDrone commented:
This PR looks fantastic, let's merge it!
Prepared branch [14.0-dev-ocabot-merge-pr-193-by-ivs-cetmix-bump-nobump](https://github.com/cetmix/cetmix-tower/commits/14.0-dev-ocabot-merge-pr-193-by-ivs-cetmix-bump-nobump), awaiting test results.

CetmixGitDrone commented:
Congratulations, your PR was merged at 7c3dde6f98062f49cf3dbf722d9d3d1809ebcfb2. Thanks a lot for contributing to cetmix. ❤️
